Sources/Sentry/SentrySessionReplayIntegration.m

```diff
@@ -46,6 +47,8 @@ @implementation SentrySessionReplayIntegration {
     SentryReplayOptions *_replayOptions;
     SentryNSNotificationCenterWrapper *_notificationCenter;
     SentryOnDemandReplay *_resumeReplayMaker;
+    id<SentryRateLimits> _rateLimits;
+    BOOL _rateLimited;
```

Comment:
`h`: What stops us from checking the rate limit for every session replay we start? Checking the rate active limits is cheap. This flag can quickly get out of sync with the rate limit, leading to potentially dropping data even if we don't need to.

Reply:
> This flag can quickly get out of sync with the rate limit

This is the intention. Once the app was rate limited, we should not start the session replay again for the entire session.

Reply:
If we always limit this to an entire session, we lose flexibility. Why can't the server simply send a long enough timeout? Why does it have to be an entire session?

Reply:
Right now, this is the easiest way to ensure segment 0 will always reach the server, because session replay absolutely needs segment 0 to make replay work. 

We could add more steps to check whether the envelope carrying the segment 0 was delivered, but it seems an overkill for an edge case (rate limiting).

Reply:
Please add this essential info as a code comment somewhere, because the future me will forget and get confused again.